File: java/DeepSpace/src/deepspace/Dice.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;
import java.util.Random;

/**
 *
 * @author mariorgzlpz
 */
public class Dice {
    private final float NHANGARSPROB=0.25f;
    private final float NSHIELDSPROB=0.25f;
    private final float NWEAPONSPROB=0.25f;
    private final float FIRSTSHOTPROB=0.25f;
    private Random generator=new Random();
    
    int initWithNHangars(){
        if(generator.nextFloat() < NHANGARSPROB)
            return 0;
        else
            return 1;
    }
    
    int initWithNWEAPONS(){
        float weaponprob = generator.nextFloat();
        if(weaponprob < NWEAPONSPROB)
            return 1;
        else if(weaponprob < 2*NWEAPONSPROB && weaponprob >= NWEAPONSPROB)
            return 2;
        else if(weaponprob >= 2*NWEAPONSPROB)
            return 3;
    }

    int initWithNShields(){
        if(generator.nextFloat() < NSHIELDSPROB)
            return 0;
        else
            return 1;
    }
    
    int whoStarts(int nPlayers){
        return generator.nextInt(nPlayers);
    }
    
    GameCharacter firstShot(){
        if(generator.nextFloat() < FIRSTSHOTPROB)
            return GameCharacter.SPACESTATION;
        else
            return GameCharacter.ENEMYSTARSHIP;
    }
    
    boolean spaceStationMoves(float speed){
        if(generator.nextFloat() < speed)
            return true;
        else
            return false;
    }
}

File: ruby/PRACTICA 2/Dice.rb
require_relative 'GameCharacter'
module Deepspace
    class Dice
        def initialize()
            @NHANGARSPROB=0.25
            @NSHIELDSPROB=0.25
            @NWEAPONSPROB=0.33
            @FIRSTSHOTPROB=0.5
            @generator=Randow.new()
        end
        
        def initWithNHangars()
            if @generator.rand()<@NHANGARSPROB
                return 0
            else
                return 1
            end
        end

        def initWithNShields()
            if @generator.rand()<@NSHIELDSPROB
                return 0
            else
                return 1
            end
        end

        def initWithNWeapons()
            if @generator.rand()<@NWEAPONSPROB
                return 0
            else
                return 1
            end
        end

        def whoStarts(nPlayers)
            return @generator.rand(nPlayers)
        end

        def firstShot()
            if @generator.rand() < @FIRSTSHOTPROB
                return GameCharacter::SPACESTATION
            else
                return GameCharacter::ENEMYSTARSHIP
            end
        end

        def spaceStationMoves(speed)
            if @generator.rand() < speed
                return true
            else
                return false
            end
        end
    end
end